lstchain/reco/dl1_to_dl2.py

```diff
@@ -517,10 +547,37 @@ def get_expected_source_pos(data, data_type, config):
         expected_src_pos_y_m = expected_src_pos.y.to_value()
 
     # For real data
-    # TODO: expected source position for real data should be obtained by using tel_alt,az and source RA,Dec
-    # For the moment, expected source position is defined as camera center (ON mode)
     if data_type == 'real_data':
-        expected_src_pos_x_m = np.zeros(len(data))
-        expected_src_pos_y_m = np.zeros(len(data))
-
+        # source is always at the ceter of camera for ON mode
+        if config['observation_mode'] == 'on':
+            expected_src_pos_x_m = np.zeros(len(data))
+            expected_src_pos_y_m = np.zeros(len(data))
+        
+        # compute source position in camera coordinate event by event for wobble mode
+        if config['observation_mode'] == 'wobble':
+
+            if 'source_name' in config:
+                source_coord  = SkyCoord.from_name(config['source_name'])
+            else:
+                source_coord  = SkyCoord(config['source_ra'], config['source_dec'], frame="icrs", unit="deg")
+     
+            focal_length = OpticsDescription.from_name('LST').equivalent_focal_length
+            
+            expected_src_pos_x_m = np.zeros(len(data))
+            expected_src_pos_y_m = np.zeros(len(data))
+            
+            for i in range(len(data)):
```

Comment:
I suppose this whole loop will get removed with the faster transformations?
It should not be necessary to iterate one event at at time, the transformations works on arrays/quantities.

Reply:
Indepentent of faster or not faster transforms. This loop must be removed, as it will slow down the code quite dramatically.

Reply:
I fully agree, removed the loop.